### PR TITLE
feat (customer portal): add customer portal auth and setup invoices query

### PR DESCRIPTION
**Pull request by lovrocolic**
## Context

Customer portal allows end user to visit basic informations like subscription and invoices

## Description

This PR handles:
- URL signature validation
- adding invoices graphQL query

### CODE REVIEW

File: app/controllers/graphql_controller.rb [hunk 1]
```diff
@@ -23,6 +24,7 @@ def execute
     context = {
       current_user: current_user,
       current_organization: current_organization,
+      customer_portal_user: customer_portal_user,
```
Comment (jdenquin) [inline app/controllers/graphql_controller.rb:27]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/graphql/resolvers/customer_portal/invoices_resolver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  before do
+    draft_invoice
+    finalized_invoice
+  end
+
+  it 'returns a list of invoices' do
+    result = execute_graphql(
+      customer_portal_user: customer,
+      query: query,
```
Comment (jdenquin) [inline spec/graphql/resolvers/customer_portal/invoices_resolver_spec.rb:31]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/graphql/resolvers/customer_portal/invoices_resolver_spec.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    )
+
+    invoices_response = result['data']['customerPortalInvoices']
+
+    aggregate_failures do
+      expect(invoices_response['collection'].count).to eq(2)
+      expect(invoices_response['collection'].pluck('id')).to match_array(
```
Comment (jdenquin) [inline spec/graphql/resolvers/customer_portal/invoices_resolver_spec.rb:38]: RSpec/MatchArray: Prefer `contain_exactly` when matching an array literal. (https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/MatchArray)

File: spec/graphql/resolvers/customer_portal/invoices_resolver_spec.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            metadata { currentPage, totalCount }
+          }
+        }
+      GQL
+    end
+
+    it 'only returns draft invoice' do
+      result = execute_graphql(
+        customer_portal_user: customer,
+        query: query,
```
Comment (jdenquin) [inline spec/graphql/resolvers/customer_portal/invoices_resolver_spec.rb:61]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/graphql/resolvers/customer_portal/invoices_resolver_spec.rb [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        expect(invoices_response['collection'].first['id']).to eq(draft_invoice.id)
+        expect(invoices_response['metadata']['totalCount']).to eq(1)
+      end
+    end
+  end
+
+  context 'without customer portal user' do
+    it 'returns an error' do
+      result = execute_graphql(
+        query: query,
```
Comment (jdenquin) [inline spec/graphql/resolvers/customer_portal/invoices_resolver_spec.rb:78]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/requests/graphql_spec.rb [hunk 6]
```diff
@@ -114,5 +114,32 @@
...
+            }
+          }
+        GQL
+      end
+      let(:token) do
+        ActiveSupport::MessageVerifier.new(ENV['SECRET_KEY_BASE']).generate(customer.id, expires_in: 12.hours)
+      end
+
+      it 'retrieves the correct end user and returns success status code' do
+        post '/graphql', headers: {
```
Comment (jdenquin) [inline spec/requests/graphql_spec.rb:135]: Layout/MultilineMethodArgumentLineBreaks: Each argument in a multi-line method call must start on a separate line.

File: spec/requests/graphql_spec.rb [hunk 7]
```diff
@@ -114,5 +114,32 @@
...
+          'customer-portal-token' => token,
+        }, params: {
+          query: query,
```
Comment (jdenquin) [inline spec/requests/graphql_spec.rb:138]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/support/graphql_helper.rb [hunk 8]
```diff
@@ -7,12 +7,13 @@ def controller
...
   end
 
-  def execute_graphql(current_user: nil, query: nil, current_organization: nil, **kwargs)
+  def execute_graphql(current_user: nil, query: nil, current_organization: nil, customer_portal_user: nil, **kwargs)
     args = kwargs.merge(
       context: {
         controller: controller,
         current_user: current_user,
         current_organization: current_organization,
+        customer_portal_user: customer_portal_user,
```
Comment (jdenquin) [inline spec/support/graphql_helper.rb:16]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

**Review by vincent-pochet — APPROVED** [review]

**Pull request merged**